### PR TITLE
tests: implement a partial DAG verifier for model orders

### PR DESCRIPTION
**Pull request by isidentical**

### CODE REVIEW

File: integration_tests/features/steps/fal_steps.py [hunk 1]
```diff
@@ -188,33 +188,72 @@ def check_model_results(context):
...
 
-LAYOUT_PARSER = re.compile(
-    r"\d+\."
-    r" (?P<subjects>[\w.]+(, [\w.]+)*)"
-)
 
-def _parse_ordered_layout(source) -> Iterator[Sequence[str]]:
-    for match in LAYOUT_PARSER.finditer(source):
-        yield set(map(_script_filename, match["subjects"].split(", ")))
+@then("the following models are calculated in order")
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:192]: this means we keep around the one that does not check order, right? `@then("the following models are calculated")`

File: integration_tests/features/steps/fal_steps.py [hunk 2]
```diff
@@ -188,33 +188,72 @@ def check_model_results(context):
...
+    from fal import FalDbt
+    from fal.node_graph import NodeGraph
 
-    unittest.TestCase().assertEqual(
-        ordered_nodes,
-        []
+    fal_dbt = FalDbt(
+        profiles_dir=_set_profiles_dir(context), project_dir=context.base_dir
     )
+    node_graph = NodeGraph.from_fal_dbt(fal_dbt)
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:208]: super nice!

File: integration_tests/features/steps/fal_steps.py [hunk 3]
```diff
@@ -188,33 +188,72 @@ def check_model_results(context):
...
+        ]
+
+    for node in ordered_nodes:
+        if _is_script(node):
+            continue
+
+        # Ancestors must precede the node
+        for ancestor in node_to_ancestors[node]:
+            _assert_precedes(ordered_nodes, ancestor, node)
+            # Post-hooks of ancestors also must precede the node
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:233]: great

File: integration_tests/features/steps/fal_steps.py [hunk 4]
```diff
@@ -188,33 +188,72 @@ def check_model_results(context):
...
+            for post_hook in node_to_post_hook[ancestor]:
+                _assert_precedes(ordered_nodes, post_hook, node)
+
+        # The node itself must precede the post-hooks
```
Comment (chamini2) [inline integration_tests/features/steps/fal_steps.py:237]: cool

**Review by chamini2 — APPROVED** [review]
Awesome!!

File: integration_tests/features/steps/fal_steps.py (hunk 1 above)
Comment (isidentical) [reply integration_tests/features/steps/fal_steps.py:192]: Yep. I initially thought about verifying everything, but then it became a bit excessive (in terms of time) and problematic on some scenarios (e.g., you might be using a different `profile_target` on the CLI etc.). So for reducing the CI time & overall complexity, this is enabled on-demand.

**Comment by isidentical** [comment]
Thanks for the review @chamini2! ❤️

**Pull request merged**